### PR TITLE
Micro-optimizations of using collections

### PR DESCRIPTION
**Pull request by danicheg**
Some optimizations to silence the $IDE linter mostly.

### CODE REVIEW

File: versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala [hunk 1]
```diff
@@ -77,10 +77,7 @@ object TypelevelVersioningPlugin extends AutoPlugin {
           .flatMap { previous =>
             if (previous > baseV)
               sys.error(s"Your tlBaseVersion $baseV is behind the latest tag $previous")
-            else if (baseV.isSameSeries(previous))
-              Some(previous)
-            else
-              None
+            else Some(previous).filter(baseV.isSameSeries)
```
Comment (armanbilge) [inline versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala:80]: I wonder if this one is more confusing now.

**Review by armanbilge — APPROVED** [review]
One thought, otherwise nice cleanups 👍 thank you! Or thank your linter for me 😉

File: versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala (hunk 1 above)
Comment (danicheg) [reply versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala:80]: I could advise another variant:
```scala
.flatMap { 
  case previous if previous > baseV =>
     sys.error(s"Your tlBaseVersion $baseV is behind the latest tag $previous")
  case previous =>
    Option(previous).filter(baseV.isSameSeries)
}
```
I often use this `Option(foo).filter(_ => ???)` pattern. But if it confuses you, then let's undo it. I'm just not so much like nested ifs tbh.

Comment (armanbilge) [reply versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala:80]: Yeah for some reason `Option(foo).filter(_ => ???)` confuses me here 😅 there are 3 cases, and it sort of merges 2 of them into one case.

I would be okay with this. It's basically the same as the if-else.
```scala
case previous if previous > baseV =>
   sys.error(s"Your tlBaseVersion $baseV is behind the latest tag $previous")
case previous if baseV.isSameSeries(previous) =>
  Some(previous)
case _ => None 
```

Comment (danicheg) [reply versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala:80]: Roger that 👍🏻

**Comment by armanbilge** [comment]
A bona-fide regular contributor! 🎉

**Pull request merged**